src/com/xilinx/rapidwright/edif/EDIFNetlist.java

```diff
@@ -553,7 +553,7 @@ private EDIFCell copyCellAndSubCellsWorker(EDIFCell cell, Set<EDIFCell> copiedCe
 			if(cell.getLibrary().isHDIPrimitivesLibrary()){
 				destLib = getHDIPrimitivesLibrary();
 			}else{
-				destLib = getWorkLibrary();
+				destLib = addLibrary(new EDIFLibrary(cell.getLibrary().getName()));
```

Comment:
There might need to be a check here to see if the destination netlist already has the same named library.  If it does, we should just add the cell to it.

Reply:
That does get checked here a few lines up:

https://github.com/Xilinx/RapidWright/blob/c2770a77b7481745abc246b3f6cad61d722bc70f/src/com/xilinx/rapidwright/edif/EDIFNetlist.java#L551-L552

Reply:
Sorry, I missed that.  Thanks!